### PR TITLE
Issues caused by missing items in hook dependency arrays

### PR DESCRIPTION
**Pull request by zpchavez**
Hi, I was running into this issue and found it's a simple fix. I was trying to restart a timer with a new initialTime and the timer would start with the old value instead. While implementing the fix I noticed the same issue with `onTimeUpdate`. Tests included.

### CODE REVIEW

**Review by thibaultboursier — COMMENTED** [review]
Hello @zpchavez,

Thanks for your PR 👍 

Could you provide, please, a [StackBlitz](https://stackblitz.com/) link in order to reproduce your issue with `initialTime`?

**Comment by zpchavez** [comment]
@thibaultboursier Sure. How's this? https://stackblitz.com/edit/react-pecbfx?file=src/App.js

**Comment by thibaultboursier** [comment]
> https://stackblitz.com/edit/react-pecbfx?file=src/App.js

Thanks @zpchavez. I'm going to check that.

File: src/useTimer.test.tsx [hunk 1]
```diff
@@ -61,6 +61,37 @@ describe('Start', () => {
...
+    // When
+    fireEvent.click(getByRole('button'));
+
+    act(() => {
+      jest.advanceTimersByTime(5000);
+    });
+
+    // Then
+    expect(getByTestId('time').textContent).toBe('25');
+  });
```
Comment (thibaultboursier) [inline src/useTimer.test.tsx]: I think that this test is not complete. Here, You just start the timer only once, after the second rendering. 
If I understood well, according to [your example](https://stackblitz.com/edit/react-pecbfx?file=src/App.js), what you want to test is:
- Start timer with a first initial value.
- Change this initial value and re-render component.
- Reset timer.
- Start timer and check that new initial value is correctly displayed.

You can check my suggestion. I just added some little changes to your test.

```suggestion
  it('should re-start timer with updated initialTime', () => {
    // Given
    const Component: React.FC<Partial<Config>> = ({ initialTime }) => {
      const { time, start, reset } = useTimer({
        initialTime,
      });

      return (
        <div>
          <button data-testid="start" onClick={start}>
            Start
          </button>
          <button data-testid="reset" onClick={reset}>
            Reset
          </button>
          <p data-testid="time">{time}</p>
        </div>
      );
    };

    const { getByTestId, rerender } = render(<Component initialTime={10} />);
    const startButton = getByTestId('start');
    const resetButton = getByTestId('reset');

    // When
    fireEvent.click(startButton);

    rerender(<Component initialTime={20} />);

    // When
    fireEvent.click(resetButton);
    fireEvent.click(startButton);

    // Then
    expect(getByTestId('time').textContent).toBe('20');
  });
```

File: src/useTimer.test.tsx [hunk 2]
```diff
@@ -627,4 +658,43 @@ describe('State and callbacks', () => {
...
+
+  it('should call updated callback function when time is updated', () => {
+    // Given
+    const initialOnTimeUpdate = jest.fn();
+    const updatedOnTimeUpdate = jest.fn();
+    const Component = ({
+      onTimeUpdate,
+    }: {
+      onTimeUpdate: (time: number) => void;
+    }) => {
```
Comment (thibaultboursier) [inline src/useTimer.test.tsx]:
```suggestion
    const Component: React.FC<Partial<Config>> = ({
      onTimeUpdate,
    }) => {
```

File: src/useTimer.test.tsx [hunk 3]
```diff
@@ -627,4 +658,43 @@ describe('State and callbacks', () => {
     expect(onTimeUpdate).toHaveBeenNthCalledWith(5, 4);
     expect(onTimeUpdate).toHaveBeenLastCalledWith(10);
   });
+
+  it('should call updated callback function when time is updated', () => {
```
Comment (thibaultboursier) [inline src/useTimer.test.tsx:669]: 👍

**Review by thibaultboursier — COMMENTED** [review]
Thanks for these improvements. Just some minor comments.

**Review by thibaultboursier — APPROVED** [review]
Perfect @zpchavez 💯 
Thanks for your work.

**Pull request merged**